block_additional_mappings/timecode.md

```diff
@@ -0,0 +1,19 @@
+## Block Additional Mapping - Timecode
+
+### Timecode Description
+
+Timecode values MAY be stored in the `BlockMore Element` to associate the content of a Matroska Block with a particular timecode value. The format of the timecode value is defined by which value is set in the BlockAddIDExtraData value.
+
+### BlockAddIDType
+
+The BlockAddIDType value reserved for timecode is 58.
+
+### BlockAddIDName
+
+The BlockAddIDName value reserved for timecode is "timecode".
+
+### BlockAddIDExtraData
+
+#### Empty
+
+If the BlockAddIDExtraData Element is empty or unstored, the format of the timecode is a string expression comprised of only the follow ASCII characters `0`, `1`, `2`, `3`, `4`, `5`, `6`, `7`, `8`, `9`, `:`, `;`, and `.`.
```

Comment:
Is there any timecode specification that would tell how to interpret these numbers ?
Is `;:25645.:111` a valid timecode ?

Reply:
I just meant this as an example and anticipated that we could use extra data to define various storage formats of timecode, but to start with one that doesn't necessarily imply rules about it. I'm thinking about how the decklink device input in ffmpeg will pass timecode as side data. The data is there as a string and there's certainly things that can be inferred about it but as side data, simply labelled as a timecode, there's no too much specific that could be stated about the storage of that data.

Perhaps we could provide a regex here though or cite SMPTE S12M, but in that case I'd suggest using the extra data to clarify that it is SMPTE S12M rather than letting the extra data be empty.

Reply:
Just like a codec, you need different `BlockAddIDType` for each timecode type you want to store. You could have multiple types of timecodes set for the same frame. They are likely incompatible. So IMO this one should be SMPTE S12M only. If there are other formats, they define they own `Block Additional Mapping`.

Now looking at [Wikipedia](https://en.wikipedia.org/wiki/SMPTE_timecode) it seems that the norm is a 32 bits value, not a string. So there should at least be a variant with the untouched 32 bits value (maybe that's the value 12 to use for `BlockAddIDType`) and one with a string, which should have a normative definition (outside of the scope of Matroska ?).